src/mlpack/methods/ann/loss_functions/poisson_nll_loss.hpp

```diff
@@ -0,0 +1,142 @@
+  typename OutputDataType = arma::mat
+>
+class PoissonNLLLoss
+{
+ public:
+  /**
+   * Create the PoissonNLLLoss object.
+   *
+   * @param logInput If true the loss is computed as exp(input) - target*input,
+   *        if false then the loss is input - target * log(input + eps).
+   * @param full Boolean value that determines whether to include Stirling's
+   *        approximation term.
+   * @param eps A small value to prevent log(0) when logInput = False.
+   * @param reduction When true, mean loss is computed otherwise total loss.
+   */
+  PoissonNLLLoss(const bool logInput = true,
+                 const bool full = false,
+                 const double eps = 1e-08,
+                 const bool reduction = true);
+
+  /**
+   * Computes the Poisson Negative log likelihood.
+   *
+   * @param input Input data used for evaluating the specified function.
+   * @param target The target vector, that contains the class index in the range
+   *        between 1 and the number of classes.
+   */
+  template<typename InputType, typename TargetType>
+  double Forward(const InputType& input, TargetType& target);
+
+  /**
+   * Ordinary feed backward pass of a neural network. The Poisson Negative Log
+   * Likelihood loss function expects the input for each class.
+   * It expects a class index, in the range between 1 and the number of classes,
+   * as target when calling the Forward function.
+   *
+   * @param input The propagated input activation.
+   * @param target The target vector, that contains the class index in the range
+   *        between 1 and the number of classes.
+   * @param output The calculated error.
+   */
+  template<typename InputType, typename TargetType, typename OutputType>
+  void Backward(const InputType& input,
+                const TargetType& target,
+                OutputType& output);
+
+  //! Get the input parameter.
+  InputDataType& InputParameter() const { return inputParameter; }
+  //! Modify the input parameter.
+  InputDataType& InputParameter() { return inputParameter; }
+
+  //! Get the output parameter.
+  OutputDataType& OutputParameter() const { return outputParameter; }
+  //! Modify the output parameter.
+  OutputDataType& OutputParameter() { return outputParameter; }
+
+  //! Get the value of logInput.
+  bool LogInput() const { return logInput; }
+  //! Modify the value of logInput.
+  bool& LogInput() { return logInput; }
+
+  //! Get the value of full.
+  bool Full() const { return full; }
+  //! Modify the value of full.
+  bool& Full() { return full; }
+
+  //! Get the value of eps.
+  double Eps() const { return eps; }
+  //! Modify the value of eps.
+  double& Eps() { return eps; }
+
+  //! Get the value of reduction.
+  bool Reduction() const { return reduction; }
+  //! Modify the value of reduction.
+  bool& Reduction() { return reduction; }
+
+  /**
+   * Serialize the layer.
+   */
+  template<typename Archive>
+  void serialize(Archive& /* ar */, const unsigned int /* version */);
```

Comment:
Could you also change `/* ar */` to `ar`, It might cause problems on older compilers (I think). Thanks.

Reply:
Hmm... I will make the changes in the next commit.